### DISCUSSION TITLE
Cleaning up Web Cache API

### DISCUSSION THREAD

**Discussion by alexgleason** [denoland/deno#18925] [General]
I'm heavily using the Web Cache API to cache results from dynamic URL queries. It needs to be cleaned up every couple days to prevent filling the entire disk, so I've added a stupid cron job that does this:

```
0 6 */2 * * systemctl stop deno && mv /tmp/deno_cache /tmp/deno_cache_old && systemctl start deno && rm -rf /tmp/deno_cache_old
```

(It takes ~20 minutes to delete the cache directory, so it's faster to move it and delete it afterwards)

I'm already adding an `Expires` header before inserting a Response into the cache. Sidenote: Cloudflare Workers clean up expired responses automatically. I'm thinking I need to emulate CF's behavior by enumerating every key in the cache and deleting expired ones, on a time interval (eg every 60 seconds). I'm a bit concerned it will be an intensive job and thinking about doing it in a Web Worker.

Anyone have experience with this?

**Comment by Ayfri** [comment]
This is a really annoying problem, if anyone from Deno could give a better solution that would be great.

**Reply by alexgleason** [reply]
I decided web cache API was not appropriate for my use case. I switched to LRUCache and never looked back. https://www.npmjs.com/package/lru-cache

**Reply by Ayfri** [reply]
Good to know, will maybe use if I have another project using Deno and SSR, thanks for the reply !

**Comment by xenova** [comment]
The current way I do it is simply by calling `await caches.delete('cache-name');` in code, but maybe there's a better way?
cc @crowlKats (as it impacts Transformers.js model caching).
